src/wallet/wallet2.cpp

```diff
@@ -2313,16 +2311,36 @@ void wallet2::load(const std::string& wallet_, const std::string& password)
 
       std::stringstream iss;
       iss << cache_data;
-      boost::archive::binary_iarchive ar(iss);
+      cereal::PortableBinaryInputArchive ar(iss);
       ar >> *this;
+      if (m_account_public_address.m_spend_public_key != m_account.get_keys().m_account_address.m_spend_public_key ||
```

Comment:
I just copied the same check in the `THROW_WALLET_EXCEPTION_IF` below. This check makes sense to me, as `m_account` is filled in `load_keys(m_keys_file, password)` at line 2279. So it checks if the public wallet address recorded in the wallet cache is consistent with the keys.